### ISSUE TITLE
Nvidia on demand crashes Cinnamon

### ISSUE THREAD

**Issue by Ziktofel** [linuxmint/cinnamon#9468]
```
 * Cinnamon version: Cinnamon 4.6.6
 * Distribution: Mint 20.0
 * Graphics hardware: Intel Corporation UHD Graphics 630 (Mobile) (prog-if 00 [VGA controller]) (i7-8750H) + NVIDIA Corporation GP106M [GeForce GTX 1060 Mobile]
 * 64 bit
 * Relevent log message:
kernel: [   85.190489] cinnamon[9709]: segfault at 28 ip 00007f16e973fae2 sp 00007fff70493700 error 6 in libmuffin.so.0.0.0[7f16e970b000+74000]
 ```

**Issue**
If monitors are connected to both GPUs and using nvidia-450 from https://launchpad.net/~graphics-drivers/+archive/ubuntu/ppa and nvidia ondemand prime mode, Cinnamon crashes

**Steps to reproduce**
0. Acquire HW with both Intel and Nvidia GPUs (preferably a laptop)
1. Install Nvidia v450 driver
2. Connect displays to both GPUs, keep all of them enabled
3. Set prime mode to ondemand
4. Reboot
5. Login with cinnamon session
6. Experience the crash

**Expected behaviour**
Everything up and running

**Other information**
Maybe muffin wrongly detects the correct driver if reverse prime (as there is a display connected to Nvidia GPU but Intel draws on it) is used which is supported since v450 (the nvidia-connected display was not available in nvidia v440 driver)

**Comment by leigh123linux** [comment]
Duplicate of #9401

**Comment by mtwebster** [comment]
Can you upload the crash info from System Reports? (preferably the crash.tar.gz file)

**Comment by Ziktofel** [comment]
Here you go [crash.tar.gz](https://github.com/linuxmint/cinnamon/files/4943345/crash.tar.gz)

**Comment by ItzSwirlz** [comment]
This was fixed in linuxmint/muffin@c092579 and is in the recent release.

This fixes regression from version 4.0.0. Onward support needs backporting.
